### PR TITLE
Make dPiece zero-indexed

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/drlg_l1.cpp [hunk 1]
```diff
@@ -567,9 +567,9 @@ void InitCryptPieces()
 {
 	for (int j = 0; j < MAXDUNY; j++) {
 		for (int i = 0; i < MAXDUNX; i++) {
-			if (dPiece[i][j] == 77) {
+			if (dPiece[i][j] == 75) {
```
Comment (glebm) [inline Source/drlg_l1.cpp:570]: 76?

File: Source/drlg_l1.cpp [hunk 2]
```diff
@@ -567,9 +567,9 @@ void InitCryptPieces()
 {
 	for (int j = 0; j < MAXDUNY; j++) {
 		for (int i = 0; i < MAXDUNX; i++) {
-			if (dPiece[i][j] == 77) {
+			if (dPiece[i][j] == 75) {
 				dSpecial[i][j] = 1;
-			} else if (dPiece[i][j] == 80) {
+			} else if (dPiece[i][j] == 78) {
```
Comment (glebm) [inline Source/drlg_l1.cpp:572]: 79?

File: Source/scrollrt.cpp [hunk 3]
```diff
@@ -960,8 +955,10 @@ void DrawFloor(const Surface &out, Point tilePosition, Point targetBufferPositio
 	}
 }
 
-#define IsWall(x, y) (dPiece[x][y] == 0 || TileHasAny(dPiece[x][y], TileProperties::Solid) || dSpecial[x][y] != 0)
-#define IsWalkable(x, y) (dPiece[x][y] != 0 && IsTileNotSolid({ x, y }))
+bool IsWall(x, y)
```
Comment (glebm) [inline Source/scrollrt.cpp]: Missing the types for `x` and `y`

File: Source/drlg_l1.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/drlg_l1.cpp:570]:
```suggestion
			if (dPiece[i][j] == 76) {
```

File: Source/drlg_l1.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/drlg_l1.cpp:572]:
```suggestion
			} else if (dPiece[i][j] == 79) {
```

File: Source/scrollrt.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/scrollrt.cpp]: Fixed

**Review by glebm — APPROVED** [review]

**Pull request merged**